spec/models/generic_object_spec.rb

```diff
@@ -34,44 +34,70 @@
     )
   end
 
-  it "should ensure presence of name" do
-    o = GenericObject.new(:generic_object_definition => definition, :name => nil)
-    expect(o).not_to be_valid
-  end
+  describe "#create" do
+    it "creates a generic object without definition and properties" do
+      expect(GenericObject.create).to be_a_kind_of(GenericObject)
+    end
 
-  it "should ensure uniqueness of name" do
-    GenericObject.new(:generic_object_definition => definition, :name => "MyGenericObject").save
-    expect(GenericObject.new(:generic_object_definition => definition, :name => "MyGenericObject")).not_to be_valid
-  end
+    it "creates a generic object with both definition and properties" do
+      expect(GenericObject.create(:max_number => max_number, :generic_object_definition => definition)).to be_a_kind_of(GenericObject)
+    end
 
-  it "should require presence of GenericObjectDefinition" do
-    expect { GenericObject.create!(:name => 'no_definition') }.to raise_error(ActiveRecord::RecordInvalid)
+    it "raises an error when create a generic object with properties but no definition" do
+      expect { GenericObject.create(:max_number => max_number) }.to raise_error(ActiveModel::UnknownAttributeError)
+    end
   end
 
   context "custom attributes" do
-    it "can be accessed as an attribute" do
-      go.save!
-      expect(go.generic_object_definition).to eq(definition)
-      expect(go.name).to                      eq(go_object_name)
-      expect(go.max_number).to                eq(max_number)
-      expect(go.server).to                    eq(server_name)
-      expect(go.flag).to                      eq(true)
-      expect(go.data_read).to                 eq(data_read)
-      expect(go.s_time.to_i).to               be_same_time_as(s_time.to_i)
-    end
+    context "with generic_object_definition" do
+      it "can be accessed as an attribute" do
+        go.save!
+        expect(go.generic_object_definition).to eq(definition)
+        expect(go.name).to                      eq(go_object_name)
+        expect(go.max_number).to                eq(max_number)
+        expect(go.server).to                    eq(server_name)
+        expect(go.flag).to                      eq(true)
+        expect(go.data_read).to                 eq(data_read)
+        expect(go.s_time.to_i).to               eq(s_time.to_i)
```

Comment:
That implies that the time is not serializing into the database properly because it's missing the microseconds.